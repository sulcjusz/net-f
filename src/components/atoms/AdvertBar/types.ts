export interface IAdvertBarProps {
  children: React.ReactNode
  className?: string
}
