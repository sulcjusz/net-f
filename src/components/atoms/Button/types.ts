export interface IButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  children: React.ReactNode
  className?: string
}
