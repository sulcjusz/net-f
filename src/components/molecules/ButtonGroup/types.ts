export interface IButtonGroupProps {
  children: React.ReactNode
}
