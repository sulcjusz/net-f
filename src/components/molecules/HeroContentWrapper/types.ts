export interface IHeroContentWrapper {
  children: React.ReactNode
}
