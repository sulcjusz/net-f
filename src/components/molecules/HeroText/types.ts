export interface IHeroTextProps {
  title: string
  children: React.ReactNode
  className?: string
}
