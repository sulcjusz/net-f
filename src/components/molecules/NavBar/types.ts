export interface INavBarProps {
  centerSlot?: React.ReactNode
  className?: string
  leftSlot?: React.ReactNode
  rightSlot?: React.ReactNode
}
