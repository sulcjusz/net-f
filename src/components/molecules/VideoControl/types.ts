export interface IVideoControlProps {
  isPlaying: boolean
  progress: number
  duration: number
  togglePlay: () => void
  handleSeek: (value: number) => void
}
