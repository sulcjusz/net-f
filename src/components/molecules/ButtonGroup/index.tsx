import React from 'react'
import {IButtonGroupProps} from './types'

const ButtonGroup: React.FC<IButtonGroupProps> = ({children}) => (
  <div className="flex space-x-4" data-testid="buttonGroup">
    {children}
  </div>
)

export default ButtonGroup
