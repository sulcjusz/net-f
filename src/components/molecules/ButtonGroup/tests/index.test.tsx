import {render, RenderResult} from '@testing-library/react'
import '@testing-library/jest-dom'
import ButtonGroup from '../index'
import {IButtonGroupProps} from '../types'

type ButtonGroupRenderProps = Partial<IButtonGroupProps>

const renderButtonGroup = (
  props?: ButtonGroupRenderProps
): RenderResult & {buttonGroup: HTMLElement} => {
  const renderResult = render(
    <ButtonGroup {...props}>
      <button>Button 1</button>
      <button>Button 2</button>
    </ButtonGroup>
  )
  const buttonGroup = renderResult.getByTestId('buttonGroup')
  return {...renderResult, buttonGroup}
}

describe('ButtonGroup component', () => {
  it('should match the snapshot', () => {
    const {buttonGroup} = renderButtonGroup()
    expect(buttonGroup.firstChild).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    const {buttonGroup} = renderButtonGroup()
    const buttons = buttonGroup.querySelectorAll('button')
    expect(buttons.length).toBe(2)
    expect(buttons[0]).toHaveTextContent('Button 1')
    expect(buttons[1]).toHaveTextContent('Button 2')
  })

  it('applies correct CSS classes', () => {
    const {buttonGroup} = renderButtonGroup()

    expect(buttonGroup).toHaveClass('flex')
    expect(buttonGroup).toHaveClass('space-x-4')
  })
})
