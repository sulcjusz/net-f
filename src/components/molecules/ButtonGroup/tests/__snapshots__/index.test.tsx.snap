// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component should match the snapshot 1`] = `
<button>
  Button 1
</button>
`;
